### PR TITLE
[BI-1386] Brapi importer: fix phenotyping importer

### PR DESCRIPTION
**Pull request by ctucker3**
# Description
**Story:** https://breedinginsight.atlassian.net/jira/software/c/projects/BI/boards/1?modal=detail&selectedIssue=BI-1386

Fixes the silent failure when the phenotypic brapi importer returns a ValidatorException. 
Add required annotations to fields in the brapi importer that are required. 

# Dependencies
bi-web: BI-1386

# Testing
Below are some files with different test scenarios that will show errors for the importer. Also included are files for the germplasm and traits importer to test that those errors still work. 

[BI_1373_TRAITS_good_traits.xls](https://github.com/Breeding-Insight/bi-api/files/8124752/BI_1373_TRAITS_good_traits.xls)
[BI-1386-bad-breeding-methods.xlsx](https://github.com/Breeding-Insight/bi-api/files/8124753/BI-1386-bad-breeding-methods.xlsx)
[BI-1386-entry-number-single-error.xlsx](https://github.com/Breeding-Insight/bi-api/files/8124755/BI-1386-entry-number-single-error.xlsx)
[BI-1386-GERMPLASM-missing-names.xls](https://github.com/Breeding-Insight/bi-api/files/8124756/BI-1386-GERMPLASM-missing-names.xls)
[BI-1386-missing-column.xlsx](https://github.com/Breeding-Insight/bi-api/files/8124757/BI-1386-missing-column.xlsx)
[BI-1386-missing-values-multiple.xlsx](https://github.com/Breeding-Insight/bi-api/files/8124758/BI-1386-missing-values-multiple.xlsx)
[BI-1386-TRAITS_bad-traits.xls](https://github.com/Breeding-Insight/bi-api/files/8124759/BI-1386-TRAITS_bad-traits.xls)
[BI-1386-TRAITS-missing-trait-name-column.xls](https://github.com/Breeding-Insight/bi-api/files/8124760/BI-1386-TRAITS-missing-trait-name-column.xls)
[BI-1386-working.xlsx](https://github.com/Breeding-Insight/bi-api/files/8124761/BI-1386-working.xlsx)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have tested my code and ensured it meets the acceptance criteria of the story
- [ ] I have create/modified unit tests to cover this change
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to documentation
- [ ] I have run TAF: _\<link to TAF run>_

### CODE REVIEW

**Review by HMS17 — APPROVED** [review]

File: src/main/java/org/breedinginsight/brapps/importer/services/processors/GermplasmProcessor.java [hunk 1]
```diff
@@ -285,12 +285,6 @@ public void getExistingBrapiData(List<BrAPIImport> importRows, Program program)
             throw new ValidatorException(validationErrors);
         }
 
-        // Check for bad breeding methods
```
Comment (nickpalladino) [inline src/main/java/org/breedinginsight/brapps/importer/services/processors/GermplasmProcessor.java:288]: Was this just left over from when the switch to using ValidatorException happened?

Comment (ctucker3) [reply src/main/java/org/breedinginsight/brapps/importer/services/processors/GermplasmProcessor.java:288]: Yep!

**Review by nickpalladino — COMMENTED** [review]
Testing seemed to work. Only thing I noticed was when testing [BI-1386-bad-breeding-methods.xlsx](https://github.com/Breeding-Insight/bi-api/files/8124753/BI-1386-bad-breeding-methods.xlsx) it would show errors but not have the option to reload the file and import again, would have to refresh whereas for [BI-1386-missing-values-multiple.xlsx](https://github.com/Breeding-Insight/bi-api/files/8124758/BI-1386-missing-values-multiple.xlsx) it would have the option there.

**Comment by ctucker3** [comment]
> Testing seemed to work. Only thing I noticed was when testing [BI-1386-bad-breeding-methods.xlsx](https://github.com/Breeding-Insight/bi-api/files/8124753/BI-1386-bad-breeding-methods.xlsx) it would show errors but not have the option to reload the file and import again, would have to refresh whereas for [BI-1386-missing-values-multiple.xlsx](https://github.com/Breeding-Insight/bi-api/files/8124758/BI-1386-missing-values-multiple.xlsx) it would have the option there.

This is something that would be a bigger fix to the brapi importer and is probably out of scope for this card. Because the missing values check happens during the initial upload you get the error right away. But, the breeding methods check happens in the background thread so the error is returned at preview time.  

I made a card for this: https://breedinginsight.atlassian.net/browse/BI-1392

**Pull request merged**